CMakeLists.txt

```diff
@@ -247,8 +247,20 @@ if (OTIO_FIND_IMATH)
         message(STATUS "Using Imath::Imath specified by ${Imath_CONFIG}")
     endif()
 else()
-    message(STATUS "Using Imath in OpenTimelineIO/src/deps")
-    set(USE_DEPS_IMATH ON)
+    if ("${ILMBASE_LIBRARIES}" STREQUAL "")
```

Comment:
I am curious, how/what is going to set this variable?

Reply:
It's used in src/deps/CMakeLists.txt, where the condition reads as below. The idea is that if Imath is provided externally, for example, if OTIO is part of a larger build, we need to respect the external Imath, and not use the one included via `add_subdirectory(Imath)`.

```
if (USE_DEPS_IMATH)
    # preserve BUILD_SHARED_LIBS options for this project, but set it off for Imath
    option(BUILD_SHARED_LIBS "Build shared libraries" ON)
    set(BUILD_SHARED_LIBS OFF)

    # If we do not want Imath to install headers and CMake files use the EXCLUDE_FROM_ALL option
    if(OTIO_CXX_INSTALL AND OTIO_DEPENDENCIES_INSTALL)
        add_subdirectory(Imath)
    else()
        add_subdirectory(Imath EXCLUDE_FROM_ALL)
    endif()
endif()
```

Reply:
`ILMBASE_LIBRARIES` is the variable we use for Imath 2, to set up the dependency:

```
if (USE_DEPS_IMATH)
    # preserve BUILD_SHARED_LIBS options for this project, but set it off for Imath
    option(BUILD_SHARED_LIBS "Build shared libraries" ON)
    set(BUILD_SHARED_LIBS OFF)

    # If we do not want Imath to install headers and CMake files use the EXCLUDE_FROM_ALL option
    if(OTIO_CXX_INSTALL AND OTIO_DEPENDENCIES_INSTALL)
        add_subdirectory(Imath)
    else()
        add_subdirectory(Imath EXCLUDE_FROM_ALL)
    endif()
endif()
```

and then finally:

```
target_link_libraries(opentimelineio 
    PUBLIC opentime ${OTIO_IMATH_TARGETS})
```

Reply:
So it `ILMBASE_LIBRARIE` would be externally provided?

Reply:
yes.

- OTIO_FIND_MATH if NO (which is the default)
    - if ILMBASE_LIBRARIES is supplied, use that
    - if not supplied, use src/deps/Imath
- else OTIO_FIND_IMATH is YES
   - use cmake's find_package
       - prefer Imath 3, because Imath 2 is deprecated
       - if Imath 3 not found, look for Imath 2
       - if Imath 2 not found, use src/deps/Imath

Reply:
we could add yet another variable ~ OTIO_IMATH_LIBRARIES ~ instead of using ILMBASE_LIBRARIES...

Reply:
@JeanChristopheMorinPerso Okay, I rewrote it a little per this conversation to make it obvious and easy to configure.

Reply:
Indeed, it's easier to read!